### PR TITLE
[KED-2911] Implement the comparison-view journey

### PR DESCRIPTION
**Pull request by tynandebold**
## Description

This PR implements the single and comparison view journey, connecting and wiring up all the necessary pieces of UI.

## Development notes

- I'm using a [Heroku-deployed server](https://apollo-server-simple-demo.herokuapp.com/graphql) to query our mock data.
- I've refactored a little bit to use the main `<Sidebar />` component for both experiment tracking and flowchart views.
- Our Apollo provider now wraps the whole app, not just the experiment-tracking view.
- There's a fair amount of prop drilling (e.g. from `<ExperimentWrapper />` down to children) though it still feels very manageable to me. Perhaps we can better leverage the Apollo cache as we go forward.
- There's a substantial rendering flash when you click on a `<RunsListCard />` and the metadata and tracking data render. I'm not sure why this is happening. Would appreciate another eye on it.
- I've created a brand new `<Switch />` component, which is used above the `<RunsList />` component to toggle between the single and comparison views.

## QA notes

Best to checkout the branch and run it locally to test everything.

NOTE: there's currently a failing Python test in the build pipeline. @AntonyMilneQB is looking into it.

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

## Legal notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: src/apollo/config.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { ApolloClient, InMemoryCache, createHttpLink } from '@apollo/client';
 
 const link = createHttpLink({
   /** our graphql endpoint */
-  uri: 'http://localhost:4142/graphql',
+  uri: 'https://apollo-server-simple-demo.herokuapp.com/graphql',
```
Comment (studioswong) [inline src/apollo/config.js:6]: would be worth adding a note / comment to remind that the actual graphql endpoint is `http://localhost:4142/graphql`

**Review by studioswong — CHANGES_REQUESTED** [review]
Thanks for this PR! the styling and UI setup looks good.

Some weird things I noticed while playing around with this branch:

- under single view, the metadata on the experiment details panel didn't seem to update on selecting different experiments - is this due to the setup of the dev server hosted on heroku? ( see below screenshot)
Screenshot 2021-11-09 at 14 11 56

- This also applies to the comparison view as well - the experiments on the details panel doesn't seem to align with the selection on the sidebar. This particularly seem to be the case when de-selecting a run on the sidebar. ( see below)
Screenshot 2021-11-09 at 14 08 47
 
Some further suggestions / comments on the UI flow:

- On first load of the page, the latest experiment ( last item in the runsList array) should be selected and loaded on default.
- Under the single view, clicking an already selected experiment on the sidebar should not render any action instead of being de-selected ( currently it will deselect that experiment with details component rendering nothing).

File: src/components/experiment-tracking/runs-list-card/index.js [hunk 2]
```diff
@@ -9,26 +9,50 @@ import './runs-list-card.css';
...
     >
-      {active && <CheckIcon className={'runs-list-card__checked'} />}
+      {(active || enableComparisonView) && (
+        <CheckIcon
+          className={classnames('runs-list-card__checked', {
+            'runs-list-card__checked--active': active,
+            'runs-list-card__checked--comparing': enableComparisonView,
+          })}
+        />
+      )}
```
[the remark this answers is not in the document]
Comment (tynandebold) [reply src/components/experiment-tracking/runs-list-card/index.js:49]: I'm not able to successfully test the `active` state. Maybe you can help me with that? I'm still learning the ins and outs of Jest.

**Review by studioswong — APPROVED** [review]
Nice one! Let's merge this in 😄

**Pull request merged**